core/common/src/main/java/alluxio/metrics/MetricsSystem.java

```diff
@@ -440,7 +440,8 @@ public static Timer timer(String name) {
    * By sending only the difference since the last RPC, this method will allow the master to
    * track the metrics for a given worker, even if the worker is restarted.
    */
-  private static List<alluxio.grpc.Metric> reportMetrics(InstanceType instanceType) {
+  private static synchronized List<alluxio.grpc.Metric> reportMetrics(InstanceType instanceType) {
```

Comment:
Why does this have to be `synchronized`?

Reply:
Because when clearing the metrics, `reportMetrics` are likely to be incorrect.

For example, we have a counter with value 4, its previous value which is stored in the LAST_REPORTED_METRICS is 3.

When the counter value is reset to 0, reportMetrics may report 0 - 3 = -3 to the leading master if we clean LAST_REPORTED_METRICS after counters.

If we clean counters after LAST_REPORTED_METRICS, it may occur that LAST_REPORTED_METRICS is 0, but counters are still 4, and delta value 4 - 0 = 4 is reported to the leading master which is also incorrect.

So when clearing the metrics, no `reportMetrics` should be going on. We can also use the same `readwriteLock` logic like `MetricsStore.clear()`: give `resetAllMetrics()` a write lock and give `reportMetrics` a read lock.

Reply:
Makes sense. Can you add a short comment in the javadoc of the method saying this is synchronized for correctness with `resetAllMetrics`

Reply:
Added, thanks!